tls.go

```diff
@@ -0,0 +1,268 @@
+	NoSNI                bool `long:"no-sni" description:"Do not send domain name in TLS Handshake regardless of whether known" json:"sni"`
+	SCTExt               bool `long:"sct" description:"Request Signed Certificate Timestamps during TLS Handshake" json:"sct"`
+
+	// TODO: Do we just lump this with Verbose (and put Verbose in TLSFlags)?
+	KeepClientLogs bool `long:"keep-client-logs" description:"Include the client-side logs in the TLS handshake"`
+
+	Time string `long:"time" description:"Explicit request time to use, instead of clock. YYYYMMDDhhmmss format."`
+	// TODO: directory? glob? How to map server name -> certificate?
+	Certificates string `long:"certificates" description:"Set of certificates to present to the server"`
+	// TODO: re-evaluate this, or at least specify the file format
+	CertificateMap string `long:"certificate-map" description:"A file mapping server names to certificates"`
+	// TODO: directory? glob?
+	RootCAs string `long:"root-cas" description:"Set of certificates to use when verifying server certificates"`
+	// TODO: format?
+	NextProtos              string `long:"next-protos" description:"A list of supported application-level protocols"`
+	ServerName              string `long:"server-name" description:"Server name used for certificate verification and (optionally) SNI"`
+	VerifyServerCertificate bool   `long:"verify-server-certificate" description:"If set, the scan will fail if the server certificate does not match the server-name, or does not chain to a trusted root."`
+	// TODO: format? mapping? zgrab1 had flags like ChromeOnly, FirefoxOnly, etc...
+	CipherSuite      string `long:"cipher-suite" description:"A list of cipher suites to use."`
+	MinVersion       int    `long:"min-version" description:"The minimum SSL/TLS version that is acceptable. 0 means that SSLv3 is the minimum."`
+	MaxVersion       int    `long:"max-version" description:"The maximum SSL/TLS version that is acceptable. 0 means use the highest supported value."`
+	CurvePreferences string `long:"curve-preferences" description:"A list of elliptic curves used in an ECDHE handshake, in order of preference."`
+	NoECDHE          bool   `long:"no-ecdhe" description:"Do not allow ECDHE handshakes"`
+	// TODO: format?
+	SignatureAlgorithms string `long:"signature-algorithms" description:"Signature and hash algorithms that are acceptable"`
+	HeartbeatEnabled    bool   `long:"heartbeat-enabled" description:"If set, include the heartbeat extension"`
+	DSAEnabled          bool   `long:"dsa-enabled" description:"Accept server DSA keys"`
+	// TODO: format?
+	ClientRandom string `long:"client-random" description:"Set an explicit Client Random (base64 encoded)"`
+	// TODO: format?
+	ClientHello string `long:"client-hello" description:"Set an explicit ClientHello (base64 encoded)"`
+}
+
+func getCSV(arg string) []string {
+	// TODO: Find standard way to pass array-valued options
```

Comment:
This is of course even more embarassing after the discussion Wednesday re: reinventing the argparse wheel; unfortunately, I don't see any standard way to get array-like inputs from the command-line.  This probably isn't the best format (and ditto for the others), but this serves as a placeholder for now.

I will go over zgrab1 again to see if I can duplicate what it was doing for these (if indeed they were configurable).

Maybe the best solution would just be to have it be configurable only in the ini?